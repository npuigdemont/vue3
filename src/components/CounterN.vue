<template>
    <div>
        <div><span class="count">{{multiplierCount}}</span></div>
        <br>
        <button @click="increment">Increment</button>
        <button @click="decrement">Decrement</button>
        <button @click="reset">Reset</button>
    </div>
</template>

<script>
export default {
    name: 'CounterN',
    props: {
        multiplier: { type: Number, default: 1},
    },
    data(){
        return {
            count: 0
        };
    },
    methods: {
        increment() {
            this.count++;
        },
        decrement(){
            this.count--;
        },
        reset() {
            this.count=0;
        }
    },
    computed: {
        multiplierCount(){
            return this.count * this.multiplier;
        }
    }
}
</script>