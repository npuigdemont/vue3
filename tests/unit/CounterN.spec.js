import { mount } from '@vue/test-utils';
import CounterN from '@/components/CounterN.vue';

describe("CounterN.vue", ()=> {
    it("renders component", ()=> {
        const wrapper = mount(CounterN);
        expect(wrapper.element).toMatchSnapshot();
    });
    it("has initial count 0", ()=> {
        const wrapper = mount(CounterN);
        expect(wrapper.vm.count).toEqual(0);
    });
    it("increment count by 1", ()=> {
        const wrapper = mount(CounterN);
        wrapper.vm.increment();
        expect(wrapper.vm.count).toEqual(1);
    });
    it("decrement count by 1", ()=> {
        const wrapper = mount(CounterN);
        wrapper.vm.decrement();
        expect(wrapper.vm.count).toEqual(-1);
    });
    it("reset count", ()=> {
        const wrapper = mount(CounterN);
        wrapper.vm.reset();
        expect(wrapper.vm.count).toEqual(0);
    });
    //testegem computed property
    it("multiplierCount is incremented", ()=> {
        const wrapper = mount(CounterN, {propsData: {multiplier: 4}});
        wrapper.vm.increment();
        expect(wrapper.vm.multiplierCount).toEqual(4);
    });
    it("multiplierCount is decremented", ()=> {
        const wrapper = mount(CounterN, {propsData: {multiplier: 4}});
        wrapper.vm.decrement();
        expect(wrapper.vm.multiplierCount).toEqual(-4);
    });
});