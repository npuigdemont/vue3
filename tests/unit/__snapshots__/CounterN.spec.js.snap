// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterN.vue renders component 1`] = `
<div>
  <div>
    <span
      class="count"
    >
      0
    </span>
  </div>
  <br />
  <button>
    Increment
  </button>
  <button>
    Decrement
  </button>
  <button>
    Reset
  </button>
</div>
`;
